[ROBO-4985] RemoteException - Client StackTrace

eduard-dumitru opened the pull request:
- Refactored `RemoteException` in `Dtos.cs` to handle stack traces differently for .NET Core and other frameworks.
- Implemented the `DivideByZero` operation in `SystemService` to test exceptions.
- Added `RemoteExceptionStackTrace_ShouldAlsoIncludeClientFrames` test to validate stack trace inclusion of client and server frames.
- Added `ShouldPartiallyContainInOrder` method in `ShouldlyHelpers.cs` for enumerable validation with logging.
- Removed unused namespaces and improved exception handling for clarity.

bogdanStan92 commented on src/UiPath.CoreIpc.Tests/Helpers/ShouldlyHelpers.cs:
```diff
@@ -114,4 +116,45 @@ private static async Task<T> Return<T>(this Task task, T value = default!)
         return value;
     }
+
+    public static void ShouldPartiallyContainInOrder(this IEnumerable<string> haystack, IEnumerable<string> needles, [CallerArgumentExpression(nameof(haystack))] string? supersetExpression = null, [CallerArgumentExpression(nameof(needles))] string? subsetExpression = null)
```
I don't see the need of all this overhead just for logging. It can be far simpler

bogdanStan92 commented on src/UiPath.CoreIpc/Wire/Dtos.cs:
```diff
@@ -66,13 +67,27 @@ public record Error(string Message, string StackTrace, string Type, Error? Inner
-        StackTrace = error.StackTrace;
+        if (error.StackTrace is not null)
+        {
+            SetRemoteStackTrace(error.StackTrace);
```
We should look into how the logs look with this change and if they help

eduard-dumitru committed:
Enhance exception handling and testing

Added new test methods in `ComputingTests.cs` to verify remote exception stack traces, including client-side and server-side frames. Introduced `ShouldPartiallyContainInOrder` in `ShouldlyHelpers.cs` for ordered string verification. Updated `ComputingCallback.cs` and `ComputingService.cs` with methods to simulate divide-by-zero exceptions. Modified interfaces in `IComputingService.cs` to support new methods. Adjusted using directives across several files for consistency and removed unused ones. Enhanced `RemoteException` stack trace handling in `Dtos.cs` for different .NET versions.

eduard-dumitru committed:
simplification
